### PR TITLE
feat(components): Introducing AWS SageMaker TrainingJob component v2 in Preview

### PR DESCRIPTION
**Pull request by surajkota**
**Description of your changes:**
- Introducing SageMaker TrainingJob pipeline component version 2 in preview. Refer to the `components/aws/sagemaker/TrainingJob/README.md` for details
- Added Sample

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: components/aws/sagemaker/Changelog.md [hunk 1]
```diff
@@ -1,9 +1,14 @@
 # Change log for AWS SageMaker Components
 
 The version of the AWS SageMaker Components is determined by the docker image tag used in YAML spec   
-Repository:  https://hub.docker.com/repository/docker/amazon/aws-sagemaker-kfp-components
+Repository: [Public ECR](https://gallery.ecr.aws/kubeflow-on-aws/aws-sagemaker-kfp-components) or [Dockerhub](https://hub.docker.com/repository/docker/amazon/aws-sagemaker-kfp-components). New releases after v1.1.1 will be using the public ECR repository
 
 ---------------------------------------------
+**Change log for version 2.0.0-alpha1**
+- Introducing SageMaker TrainingJob component v2 in Preview. This release only includes one [TrainingJob component](./TrainingJob/).
```
Comment (ryansteakley) [inline components/aws/sagemaker/Changelog.md:8]: does this mean to say "This release only includes one component... which is TrainingJob"

**Review by ryansteakley — APPROVED** [review]
Thanks for this :)

File: components/aws/sagemaker/TrainingJob/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# SageMaker Training Kubeflow Pipelines component
```
Comment (mbaijal) [inline components/aws/sagemaker/TrainingJob/README.md:1]: Has this folder structure been discussed already ? having both train and TrainingJob seems confusing.

File: components/aws/sagemaker/TrainingJob/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Component to create [SageMaker Training jobs](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-training.html) in a Kubeflow Pipelines workflow.
+
+| :warning: WARNING: This component is in Preview Release Phase. It is not GA |
+| --- |
+
+## Overview
+
+The Amazon SageMaker components for Kubeflow Pipelines version 1(v1.1.x or below) uses [Boto3 (AWS SDK for Python)](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html) as the backend to create and manage resources on SageMaker. SageMaker components version 2(v2.0.0-alpha1 or above) uses the [ACK Service Controller for SageMaker](https://github.com/aws-controllers-k8s/sagemaker-controller) to do the same. AWS introduced [ACK](https://aws-controllers-k8s.github.io/community/) to facilitate a Kubernetes-native way of managing AWS Cloud resources. ACK includes a set of AWS service-specific controllers, one of which is the SageMaker controller. The SageMaker controller makes it easier for machine learning developers and data scientists who use Kubernetes as their control plane to train, tune, and deploy machine learning models in Amazon SageMaker.
```
Comment (mbaijal) [inline components/aws/sagemaker/TrainingJob/README.md:10]: Can we link to the train/README.

**Review by mbaijal — COMMENTED** [review]
Considering the steps have been tested in bugbash, I did not verify the the guide line by line but overall looks ok once you address the comments.

**Review by mbaijal — APPROVED** [review]
Approving in case you need to merge, we can make further changes in new PRs if needed.

File: components/aws/sagemaker/TrainingJob/README.md (hunk 2 above)
Comment (surajkota) [reply components/aws/sagemaker/TrainingJob/README.md]: fair point. Added Versioning in the main README and changed the title to include v2 for now. Maybe we can use the same mechanism for other components as well

File: components/aws/sagemaker/TrainingJob/README.md (hunk 3 above)
Comment (surajkota) [reply components/aws/sagemaker/TrainingJob/README.md:10]: Changed the title and in the main README called out v1 and v2 clearly. I think it should be good for now, wdyt

File: components/aws/sagemaker/TrainingJob/samples/mnist-kmeans-training/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+numpy
```
Comment (surajkota) [inline components/aws/sagemaker/TrainingJob/samples/mnist-kmeans-training/requirements.txt:1]: pin versions

**Comment by surajkota** [comment]
@chensun is it possible to configure the tests further to run based on the directory of changes?

I cannot access the [logs](https://4e18c21c9d33d20f-dot-datalab-vm-staging.googleusercontent.com/#/runs/details/c6f96d1d-b131-4223-971e-b90a629ba245) for the samples-v2 test but also the failure would not be related to this PR. I dont think the current PR tests are adding any value to components under aws folder

Can you help us with unblocking this test failure?

**Pull request merged**